lib/Transforms/IPO/Inliner.cpp

```diff
@@ -153,6 +153,12 @@ static bool InlineCallIfPossible(CallSite CS, InlineFunctionInfo &IFI,
 
   AdjustCallerSSPLevel(Caller, Callee);
 
+  // HLSL Change Begin- not merge allocas.
+  // Merge alloca will make alloca which has one def become multi def.
+  // SROA will fail to remove the merged allocas.
+  return true;
```

Comment:
I'd expect this to cause an unreachable code warning / error. Better `#if 0` the rest of the function or somehow test that this is HLSL.